packages/form-js-viewer/src/render/components/form-fields/Number.js

```diff
@@ -11,31 +13,137 @@ import {
   prefixId
 } from '../Util';
 
+import {
+  isNullEquivalentValue,
+  isValidNumber,
+  willKeyProduceValidNumber
+} from '../util/numberFieldUtil';
+
 const type = 'number';
 
 
-export default function Number(props) {
+export default function Numberfield(props) {
   const {
     disabled,
     errors = [],
     field,
-    value
+    value,
+    onChange
   } = props;
 
   const {
     description,
     id,
     label,
-    validate = {}
+    validate = {},
+    decimalDigits,
+    serializeToString = false,
+    increment
   } = field;
 
   const { required } = validate;
 
-  const onChange = ({ target }) => {
-    props.onChange({
-      field,
-      value: Number.sanitizeValue({ value: target.value })
-    });
+  const inputRef = useRef();
+
+  const [ stringValueCache, setStringValueCache ] = useState('');
+
+  const valueCacheMismatch = useMemo(() => Numberfield.sanitizeValue({ value, formField: field }) !== Numberfield.sanitizeValue({ value: stringValueCache, formField: field }), [ stringValueCache, value, field ]);
+
+  const displayValue = useMemo(() => {
+
+    if (value === 'NaN') return 'NaN';
+    return valueCacheMismatch ? ((value || value === 0) ? Big(value).toFixed() : '') : stringValueCache;
+
+  }, [ stringValueCache, value, valueCacheMismatch ]);
+
+  const arrowIncrement = useMemo(() => {
+
+    if (increment) return Big(increment);
+    if (decimalDigits) return Big(`1e-${decimalDigits}`);
+    return Big('1');
+
+  }, [ decimalDigits, increment ]);
+
+
+  const setValue = useCallback((stringValue) => {
+
+    if (isNullEquivalentValue(stringValue)) {
+      setStringValueCache('');
+      onChange({ field, value: null });
+      return;
+    }
+
+    // treat commas as dots
+    stringValue = stringValue.replaceAll(',', '.');
+
+    if (isNaN(Number(stringValue))) {
+      setStringValueCache('NaN');
+      onChange({ field, value: 'NaN' });
+      return;
+    }
+
+    setStringValueCache(stringValue);
+    onChange({ field, value: serializeToString ? stringValue : Number(stringValue) });
+
+  }, [ field, onChange, serializeToString ]);
+
+  const addIncrement = () => {
```

Comment:
```suggestion
  const increment = () => {
```

Just to be consistent

Reply:
Had to shift some namings around for this one but this is done.